diff --git a/src/plugins/js/lock-file/lock-file.js b/src/plugins/js/lock-file/lock-file.js
index 8381f6c85e2cadc6c8c18452cfb8f27c28854cf1..4599446aefae9ed08827e79d021d35106ce2e91f 100644
--- a/src/plugins/js/lock-file/lock-file.js
+++ b/src/plugins/js/lock-file/lock-file.js
@@ -133,9 +133,18 @@ function getLockFileName(packageManager) {
     if (packageManager === 'npm') {
         return NPM_LOCK_FILE;
     }
-    if (packageManager === 'bun') {
-        return BUN_LOCK_FILE;
-    }
+    
+  if (packageManager === 'bun') {
+    try {
+      const bunVersion = (0, node_child_process_1.execSync)('bun --version').toString().trim();
+      if ((0, semver_1.gte)(bunVersion, '1.2.0')) {
+        return BUN_TEXT_LOCK_FILE;
+      }
+      return BUN_LOCK_FILE;
+    } catch {
+      return BUN_LOCK_FILE;
+    }
+  }
     throw new Error(`Unknown package manager: ${packageManager}`);
 }
 function getLockFilePath(packageManager) {
